src/server/models/ProjectModel.ts

```diff
@@ -14,11 +14,14 @@ interface ProjectModel {
 }
 
 export interface IProjectModel extends ProjectModel {
-    categories: ICategoryModel[];
+    categories?: ICategoryModel[];
 }
 
-export const Projects: Sequelize.Model<undefined, ProjectModel> =
-    sequelize.define<undefined, ProjectModel>('projects', {
+interface IProjectInstance extends Sequelize.Instance<IProjectModel> {
```

Comment:
I think sequelize won't ever populate the categories attribute  the way this is implying. You project instance should just extend projectmodel